### DISCUSSION TITLE
Response Rules: Body values

### DISCUSSION THREAD

**Discussion by roybarber** [mockoon/mockoon#502] [Help] [answered]
Hi Everyone, great product.

I have minor question, its probably something I'm doing wrong. But I have the below body being sent to my endpoint:

```
{
    "step":0,
    "submit":null
}
```

I want to be able to serve different responses based on the step value. But if I setup a rule, then nothing is happening. 

This is what my rule looks like:
![image](https://user-images.githubusercontent.com/2923782/121202775-77183e00-c86d-11eb-9546-a8a838dfc21c.png)

So my response should fail, but instead it returns a 200 with the json. 

In my son response body I can put: 

```
{
	"bodyStep": "{{ body 'step' }}"
}
```

This is returning the value of 0

Any tips or pointers as to what I'm doing wrong?

**Comment by 255kb** [comment, marked as answer]
Hi @roybarber 

There is one thing to understand about responses/rules that is not obvious (sorry for that): the first response is always the default one. So, if no rule matches, the first response in the list will always be served. I suggest that you put a 403/404 or another error at the top of the list. You can drag and drop responses to reorder them, so it can takes precedence. 
More information on rules can be found [here](https://mockoon.com/docs/latest/route-responses/multiple-responses/#default-route-response). 
We plan to change this system soon, see #252. The drag and drop reorder will stay as it also defines in which order rules will get interpreted, but the default one will be defined by a flag (there is currently a flag that is not movable).
It's always a balance between more features/customizability and easy to use UI 🙈.

**Reply by roybarber** [reply]
So simple! Swapped the 500 to the top and bingo it works. Thank you so much @255kb coffee coming your way...

**Reply by roybarber** [reply]
@255kb is there a way to do a rule for if an attribute = null

**Reply by 255kb** [reply]
Parameters that are present but empty will get parsed as empty strings by the server. Which means that you can enable regex for a rule and use something like this `^$`. However, it's not possible yet to test for a parameter (header, etc.) complete absence. There is an open issue for this: #255 

Thank you for your support, it motivates us even more 🥳
